attribute-service-api/src/main/proto/org/hypertrace/core/attribute/service/v1/attribute_metadata.proto

```diff
@@ -137,3 +138,39 @@ message AttributeMetadataFilter {
 message Empty {
 }
 
+message AttributeDefinition {
+  oneof value {
+    string span_path = 1;
+    Projection projection = 2;
+  }
+}
+
+message Projection {
+  oneof value {
+    string attribute_key = 1;
+    LiteralValue literal = 2;
+    ProjectionExpression expression = 3;
+  }
+}
+
+message LiteralValue {
+  oneof value {
+    string string_value = 1;
+    bool boolean_value = 2;
+    double float_value = 3;
+    int64 int_value = 4;
+  }
+}
+
+message ProjectionExpression {
+  ProjectionOperator operator = 1;
+  repeated Projection arguments = 2;
```

Comment:
Do we want this to be variable arity? Could do most things with chaining, but more confusing

Reply:
I think we should keep it.